src/istio/chart/templates/gateway.yaml

```
@@ -32,7 +32,8 @@ spec:
         mode: {{ $server.mode }}
         {{- if ne $server.mode "PASSTHROUGH" }}
         credentialName: gateway-tls
-        minProtocolVersion: TLSV1_3
+        # if supportTLSV1_2 is both defined and true, use TLSV1_2, otherwise use TLSV1_3
+        minProtocolVersion: {{ if (and $.Values.tls.supportTLSV1_2 (hasKey $.Values.tls "supportTLSV1_2")) }}TLSV1_2{{ else }}TLSV1_3{{ end }}
```

Do you happen to have a set of values that I could use to reproduce the issue mentioned [here](https://github.com/defenseunicorns/uds-core/pull/624#issuecomment-2261567370)? Typically with helm you are safe to just use a conditional as long as the parent key(s) are defined - if the specific key (`supportTLSV1_2`) doesn't exist it will evaluate as false, which should work in our favor here.

It does fail with a helm `ternary` but with just if/else it appears to work even after dropping all reference to `supportTLSV1_2` in the values files (validated `helm template` against this chart, passing in each of the config- values files. Ternary would be nicer but this is slightly easier to handle the conditional with I think.

```suggestion
        minProtocolVersion: {{ if $.Values.tls.supportTLSV1_2 }}TLSV1_2{{ else }}TLSV1_3{{ end }}
```

---

@mjnagel so I very well could be wrong w/ how this behaved. It was my first time setting up the dev stuff in the repo and I ran into a few problems along the way. Let experiment again with the above and i'll report back!

---

So this was silly - I had originally tried to use:

```
minProtocolVersion: {{ if .Values.tls.supportTLSV1_2 }}TLSV1_2{{ else }}TLSV1_3{{ end }}
```

And I got this error with it:

```
      WARNING  Attempt (1/3) failed with: template: uds-istio-config/templates/gateway.yaml:36:41:
               executing "uds-istio-config/templates/gateway.yaml" at <.Values.tls.supportTLSV1_2>: nil
               pointer evaluating interface {}.tls
```

But when I changed it to be prefixed with the `$` it worked as expected (`$.Values`). Anyway this is all pushed up and 🤞 that tests go through and such!